### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by emhamza**
### In the first milestone of this project I have accomplished the mentioned tasks

- [x] Config Linters for ruby project
- [x] Add a Person class
- [x] Add a Student class
- [x] Add a Teacher class in separate files

### CODE REVIEW

**Review by AlphaNtihinduka — COMMENTED** [review]
Hi @emhamza ,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
# Invalid Code Review Request ❌ 

You have submitted a project with a readme file that is not detailed. Kindly use the provided [Microverse Readme template](https://github.com/microverseinc/readme-template), edit and customize your readme with the project information for it to be professional.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready pray

Cheers and Happy coding! 👏 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1 @@
+gem 'rubocop', '>= 1.0', '< 2.0'
```
Comment (christianbiring1) [inline Gemfile:3]: - [x] Kindly make sure you have an explicit global source for your `gem`. You may consider adding this on the top of your `Gemfile`.
```
source "https://rubygems.org"
```
image

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(classroom:, age:, parent_permission: true, name: 'Unknown')
+    super(age: age, name: name, parent_permission: parent_permission)
+    @classroom = classroom
+  end
+
+  def play_hooky
+    '¯\(ツ)/¯'
```
Comment (christianbiring1) [inline student.rb]: - [x] Kindly make sure you have set your class properly as you may notice how it's not possible to create an instance of your class
image
```suggestion
class Student < Person
  attr_accessor :classroom

  def initialize(age, classroom, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @classroom = classroom
  end

  def play_hooky
    '¯\(ツ)/¯'
```

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age:, name: 'Unknown', parent_permission: true)
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (christianbiring1) [inline person.rb]: - [x] Kindly make sure you use the correct syntax while initializing the param of your class. as you may notice how it's impossible to create the instance of your classes in the `irb`
- [x] Also make sure you have an id generator for each instance of your class.
```suggestion
def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
```

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age:, specialization:, name: 'Unknown', parent_permission: true)
+    super(age: age, name: name, parent_permission: parent_permission)
+    @specialization = specialization
+  end
```
Comment (christianbiring1) [inline teacher.rb]: - [x] Kindly make sure you use the correct syntax while initializing the param of your class. as you may notice how it's impossible to create the instance of your classes in the irb
```suggestion
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @emhamza , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 

## Highlights
- [ ] Great work on implementing the classes as required in the requirements :+1: 
- [ ] Descriptive PR `title` and `summary` :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile (hunk 1 above)
Comment (emhamza) [reply Gemfile:3]: @christianbiring1 Thank you for pointing that out,  I will certainly update my Gemfile 💯

File: student.rb (hunk 2 above)
Comment (emhamza) [reply student.rb]: For sure, thank you so much for your progressive feedback 🥇

File: teacher.rb (hunk 4 above)
Comment (emhamza) [reply teacher.rb]: thank you for you progressive feedback once again! 💯

File: person.rb (hunk 3 above)
Comment (emhamza) [reply person.rb]: @christianbiring1 I appreciate your feedback 👍 But I am afraid `parent_permission = true` will create a linters' error. And the rubocop suggested to use `parent_permission: true` instead. Here is the demonstration from rubocop:
  
error

**Pull request merged**